From 4058a6a3a985df8b9a098dddc16be03a9d49cf9c Mon Sep 17 00:00:00 2001
From: sandhya <sandhyax.sebastian@intel.com>
Date: Thu, 18 Apr 2013 15:54:10 +0530
Subject: framework/av/media build error after FORTIFY_SOURCE enabled for x86

BZ: 94038

This build error is because of src and dest of memcpy are same.
In some cases same src and dest pointers are passed to memcpy.
Added a check so that if the src and dest are same then the
memcpy will not be invoked.

Category: aosp improvement
Domain: AUDIO_MEDIA-Framework
Origin: Internal
Upstream-Candidate: no

Change-Id: I5b3dfaff8500ac4639c3037d621c47cfa7f9da4c
Orig-Change-Id: I5dc5cc39ac25754ff3b01cc3aa2fa87722eb50be
Signed-off-by: sandhya <sandhyax.sebastian@intel.com>
---
 media/libeffects/downmix/EffectDownmix.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/media/libeffects/downmix/EffectDownmix.c b/media/libeffects/downmix/EffectDownmix.c
index f17a6e8..89035aa 100644
--- a/media/libeffects/downmix/EffectDownmix.c
+++ b/media/libeffects/downmix/EffectDownmix.c
@@ -629,7 +629,9 @@ int Downmix_Configure(downmix_module_t *pDwmModule, effect_config_t *pConfig, bo
         return -EINVAL;
     }
 
-    memcpy(&pDwmModule->config, pConfig, sizeof(effect_config_t));
+    if(&pDwmModule->config != pConfig) {
+        memcpy(&pDwmModule->config, pConfig, sizeof(effect_config_t));
+    }
 
     if (init) {
         pDownmixer->type = DOWNMIX_TYPE_FOLD;
-- 
1.7.4.1

